src/System.ComponentModel.Primitives/src/System/ComponentModel/CategoryAttribute.cs

```diff
@@ -0,0 +1,335 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System.ComponentModel
+{
+    /// <devdoc>
+    ///    <para>Specifies the category in which the property or event will be displayed in a
+    ///       visual designer.</para>
+    /// </devdoc>
+    [AttributeUsage(AttributeTargets.All)]
+    public class CategoryAttribute : Attribute, IIsDefaultAttribute
+    {
+        private static volatile CategoryAttribute s_appearance;
+        private static volatile CategoryAttribute s_asynchronous;
+        private static volatile CategoryAttribute s_behavior;
+        private static volatile CategoryAttribute s_data;
+        private static volatile CategoryAttribute s_design;
+        private static volatile CategoryAttribute s_action;
+        private static volatile CategoryAttribute s_format;
+        private static volatile CategoryAttribute s_layout;
+        private static volatile CategoryAttribute s_mouse;
+        private static volatile CategoryAttribute s_key;
+        private static volatile CategoryAttribute s_focus;
+        private static volatile CategoryAttribute s_windowStyle;
+        private static volatile CategoryAttribute s_dragDrop;
+        private static volatile CategoryAttribute s_defAttr;
+
+        private bool _localized;
+
+        /// <devdoc>
+        ///    <para>
+        ///       Provides the actual category name.
+        ///    </para>
+        /// </devdoc>
+        private string _categoryValue;
+
+        /// <devdoc>
+        ///    <para>Gets the action category attribute.</para>
+        /// </devdoc>
+        public static CategoryAttribute Action
+        {
+            get
+            {
+                if (s_action == null)
+                {
+                    s_action = new CategoryAttribute(nameof(Action));
+                }
+                return s_action;
```

Comment:
I assume perf isn't important here, in which case this is fine.  If there are cases where this would need to be optimized, it'd be better to only read the volatile s_action once.  You can do that and keep the code clean with a pattern like:
```C#
public static CategoryAttribute Action => s_action ?? (s_action = new CategoryAttribute(nameof(Action)));
```